warning in multiclass_nms_op.cc

```
[10:33:58]W:	 [Step 1/1] /paddle/paddle/operators/multiclass_nms_op.cc: In instantiation of 'void paddle::operators::MultiClassNMSKernel<T>::MultiClassNMS(const paddle::framework::ExecutionContext&, const Tensor&, const Tensor&, std::map<int, std::vector<int> >&, int&) const [with T = double; paddle::operators::Tensor = paddle::framework::Tensor]':
[10:33:58]W:	 [Step 1/1] /paddle/paddle/operators/multiclass_nms_op.cc:271:20:   required from 'void paddle::operators::MultiClassNMSKernel<T>::Compute(const paddle::framework::ExecutionContext&) const [with T = double]'
[10:33:58]W:	 [Step 1/1] /paddle/paddle/operators/multiclass_nms_op.cc:384:57:   required from here
[10:33:58]W:	 [Step 1/1] /paddle/paddle/operators/multiclass_nms_op.cc:204:27: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
[10:33:58]W:	 [Step 1/1]          for (int j = 0; j < label_indices.size(); ++j) {
[10:33:58]W:	 [Step 1/1]                            ^
[10:33:58]W:	 [Step 1/1] /paddle/paddle/operators/multiclass_nms_op.cc:218:25: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
[10:33:58]W:	 [Step 1/1]        for (int j = 0; j < score_index_pairs.size(); ++j) {
[10:33:58]W:	 [Step 1/1]                          ^
[10:33:58]W:	 [Step 1/1] /paddle/paddle/operators/multiclass_nms_op.cc: In instantiation of 'void paddle::operators::MultiClassNMSKernel<T>::MultiClassOutput(const Tensor&, const Tensor&, std::map<int, std::vector<int> >&, paddle::operators::Tensor*) const [with T = double; paddle::operators::Tensor = paddle::framework::Tensor]':
[10:33:58]W:	 [Step 1/1] /paddle/paddle/operators/multiclass_nms_op.cc:289:27:   required from 'void paddle::operators::MultiClassNMSKernel<T>::Compute(const paddle::framework::ExecutionContext&) const [with T = double]'
[10:33:58]W:	 [Step 1/1] /paddle/paddle/operators/multiclass_nms_op.cc:384:57:   required from here
[10:33:58]W:	 [Step 1/1] /paddle/paddle/operators/multiclass_nms_op.cc:241:25: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
[10:33:58]W:	 [Step 1/1]        for (int j = 0; j < indices.size(); ++j) {
[10:33:58]W:	 [Step 1/1]                          ^
[10:33:58]W:	 [Step 1/1] /paddle/paddle/operators/multiclass_nms_op.cc: In instantiation of 'void paddle::operators::MultiClassNMSKernel<T>::MultiClassNMS(const paddle::framework::ExecutionContext&, const Tensor&, const Tensor&, std::map<int, std::vector<int> >&, int&) const [with T = float; paddle::operators::Tensor = paddle::framework::Tensor]':
[10:33:58]W:	 [Step 1/1] /paddle/paddle/operators/multiclass_nms_op.cc:271:20:   required from 'void paddle::operators::MultiClassNMSKernel<T>::Compute(const paddle::framework::ExecutionContext&) const [with T = float]'
[10:33:58]W:	 [Step 1/1] /paddle/paddle/operators/multiclass_nms_op.cc:384:57:   required from here
[10:33:58]W:	 [Step 1/1] /paddle/paddle/operators/multiclass_nms_op.cc:204:27: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
[10:33:58]W:	 [Step 1/1]          for (int j = 0; j < label_indices.size(); ++j) {
[10:33:58]W:	 [Step 1/1]                            ^
[10:33:58]W:	 [Step 1/1] /paddle/paddle/operators/multiclass_nms_op.cc:218:25: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
[10:33:58]W:	 [Step 1/1]        for (int j = 0; j < score_index_pairs.size(); ++j) {
[10:33:58]W:	 [Step 1/1]                          ^
[10:33:58]W:	 [Step 1/1] /paddle/paddle/operators/multiclass_nms_op.cc: In instantiation of 'void paddle::operators::MultiClassNMSKernel<T>::MultiClassOutput(const Tensor&, const Tensor&, std::map<int, std::vector<int> >&, paddle::operators::Tensor*) const [with T = float; paddle::operators::Tensor = paddle::framework::Tensor]':
[10:33:58]W:	 [Step 1/1] /paddle/paddle/operators/multiclass_nms_op.cc:289:27:   required from 'void paddle::operators::MultiClassNMSKernel<T>::Compute(const paddle::framework::ExecutionContext&) const [with T = float]'
[10:33:58]W:	 [Step 1/1] /paddle/paddle/operators/multiclass_nms_op.cc:384:57:   required from here
[10:33:58]W:	 [Step 1/1] /paddle/paddle/operators/multiclass_nms_op.cc:241:25: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
[10:33:58]W:	 [Step 1/1]        for (int j = 0; j < indices.size(); ++j) {
[10:33:58]W:	 [Step 1/1]                          ^
[10:33:58]W:	 [Step 1/1] /paddle/paddle/operators/multiclass_nms_op.cc: In instantiation of 'void paddle::operators::MultiClassNMSKernel<T>::NMSFast(const Tensor&, const Tensor&, T, T, T, int64_t, std::vector<int>*) const [with T = double; paddle::operators::Tensor = paddle::framework::Tensor; int64_t = long int]':
[10:33:58]W:	 [Step 1/1] /paddle/paddle/operators/multiclass_nms_op.cc:191:14:   required from 'void paddle::operators::MultiClassNMSKernel<T>::MultiClassNMS(const paddle::framework::ExecutionContext&, const Tensor&, const Tensor&, std::map<int, std::vector<int> >&, int&) const [with T = double; paddle::operators::Tensor = paddle::framework::Tensor]'
[10:33:58]W:	 [Step 1/1] /paddle/paddle/operators/multiclass_nms_op.cc:271:20:   required from 'void paddle::operators::MultiClassNMSKernel<T>::Compute(const paddle::framework::ExecutionContext&) const [with T = double]'
[10:33:58]W:	 [Step 1/1] /paddle/paddle/operators/multiclass_nms_op.cc:384:57:   required from here
[10:33:58]W:	 [Step 1/1] /paddle/paddle/operators/multiclass_nms_op.cc:154:25: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
[10:33:58]W:	 [Step 1/1]        for (int k = 0; k < selected_indices->size(); ++k) {
[10:33:58]W:	 [Step 1/1]                          ^
[10:33:58]W:	 [Step 1/1] /paddle/paddle/operators/multiclass_nms_op.cc: In instantiation of 'void paddle::operators::MultiClassNMSKernel<T>::NMSFast(const Tensor&, const Tensor&, T, T, T, int64_t, std::vector<int>*) const [with T = float; paddle::operators::Tensor = paddle::framework::Tensor; int64_t = long int]':
[10:33:58]W:	 [Step 1/1] /paddle/paddle/operators/multiclass_nms_op.cc:191:14:   required from 'void paddle::operators::MultiClassNMSKernel<T>::MultiClassNMS(const paddle::framework::ExecutionContext&, const Tensor&, const Tensor&, std::map<int, std::vector<int> >&, int&) const [with T = float; paddle::operators::Tensor = paddle::framework::Tensor]'
[10:33:58]W:	 [Step 1/1] /paddle/paddle/operators/multiclass_nms_op.cc:271:20:   required from 'void paddle::operators::MultiClassNMSKernel<T>::Compute(const paddle::framework::ExecutionContext&) const [with T = float]'
[10:33:58]W:	 [Step 1/1] /paddle/paddle/operators/multiclass_nms_op.cc:384:57:   required from here
[10:33:58]W:	 [Step 1/1] /paddle/paddle/operators/multiclass_nms_op.cc:154:25: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
[10:33:58]W:	 [Step 1/1] /paddle/paddle/operators/multiclass_nms_op.cc: In instantiation of 'void paddle::operators::GetMaxScoreIndex(const std::vector<_RealType>&, T, int, std::vector<std::pair<T, int> >*) [with T = double]':
[10:33:58] :	 [Step 1/1] Scanning dependencies of target backward
[10:33:58]W:	 [Step 1/1] /paddle/paddle/operators/multiclass_nms_op.cc:145:21:   required from 'void paddle::operators::MultiClassNMSKernel<T>::NMSFast(const Tensor&, const Tensor&, T, T, T, int64_t, std::vector<int>*) const [with T = double; paddle::operators::Tensor = paddle::framework::Tensor; int64_t = long int]'
[10:33:58]W:	 [Step 1/1] /paddle/paddle/operators/multiclass_nms_op.cc:191:14:   required from 'void paddle::operators::MultiClassNMSKernel<T>::MultiClassNMS(const paddle::framework::ExecutionContext&, const Tensor&, const Tensor&, std::map<int, std::vector<int> >&, int&) const [with T = double; paddle::operators::Tensor = paddle::framework::Tensor]'
[10:33:58]W:	 [Step 1/1] /paddle/paddle/operators/multiclass_nms_op.cc:271:20:   required from 'void paddle::operators::MultiClassNMSKernel<T>::Compute(const paddle::framework::ExecutionContext&) const [with T = double]'
[10:33:58]W:	 [Step 1/1] /paddle/paddle/operators/multiclass_nms_op.cc:384:57:   required from here
[10:33:58]W:	 [Step 1/1] /paddle/paddle/operators/multiclass_nms_op.cc:88:27: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
[10:33:58]W:	 [Step 1/1]    if (top_k > -1 && top_k < sorted_indices->size()) {
[10:33:58]W:	 [Step 1/1]                            ^
[10:33:58]W:	 [Step 1/1] /paddle/paddle/operators/multiclass_nms_op.cc: In instantiation of 'void paddle::operators::GetMaxScoreIndex(const std::vector<_RealType>&, T, int, std::vector<std::pair<T, int> >*) [with T = float]':
[10:33:58]W:	 [Step 1/1] /paddle/paddle/operators/multiclass_nms_op.cc:145:21:   required from 'void paddle::operators::MultiClassNMSKernel<T>::NMSFast(const Tensor&, const Tensor&, T, T, T, int64_t, std::vector<int>*) const [with T = float; paddle::operators::Tensor = paddle::framework::Tensor; int64_t = long int]'
[10:33:58]W:	 [Step 1/1] /paddle/paddle/operators/multiclass_nms_op.cc:191:14:   required from 'void paddle::operators::MultiClassNMSKernel<T>::MultiClassNMS(const paddle::framework::ExecutionContext&, const Tensor&, const Tensor&, std::map<int, std::vector<int> >&, int&) const [with T = float; paddle::operators::Tensor = paddle::framework::Tensor]'
[10:33:58]W:	 [Step 1/1] /paddle/paddle/operators/multiclass_nms_op.cc:271:20:   required from 'void paddle::operators::MultiClassNMSKernel<T>::Compute(const paddle::framework::ExecutionContext&) const [with T = float]'
[10:33:58]W:	 [Step 1/1] /paddle/paddle/operators/multiclass_nms_op.cc:384:57:   required from here
[10:33:58]W:	 [Step 1/1] /paddle/paddle/operators/multiclass_nms_op.cc:88:27: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
```

--- a/paddle/operators/multiclass_nms_op.cc
+++ b/paddle/operators/multiclass_nms_op.cc
@@ -85,7 +85,7 @@ static inline void GetMaxScoreIndex(
   std::stable_sort(sorted_indices->begin(), sorted_indices->end(),
                    SortScorePairDescend<int>);
   // Keep top_k scores if needed.
-  if (top_k > -1 && top_k < sorted_indices->size()) {
+  if (top_k > -1 && top_k < static_cast<int>(sorted_indices->size())) {
     sorted_indices->resize(top_k);
   }
 }
@@ -151,7 +151,7 @@ class MultiClassNMSKernel : public framework::OpKernel<T> {
     while (sorted_indices.size() != 0) {
       const int idx = sorted_indices.front().second;
       bool keep = true;
-      for (int k = 0; k < selected_indices->size(); ++k) {
+      for (size_t k = 0; k < selected_indices->size(); ++k) {
         if (keep) {
           const int kept_idx = (*selected_indices)[k];
           T overlap = JaccardOverlap<T>(bbox_data + idx * box_size,
@@ -201,7 +201,7 @@ class MultiClassNMSKernel : public framework::OpKernel<T> {
         int label = it.first;
         const T* sdata = scores_data + label * predict_dim;
         const std::vector<int>& label_indices = it.second;
-        for (int j = 0; j < label_indices.size(); ++j) {
+        for (size_t j = 0; j < label_indices.size(); ++j) {
           int idx = label_indices[j];
           PADDLE_ENFORCE_LT(idx, predict_dim);
           score_index_pairs.push_back(
@@ -215,7 +215,7 @@ class MultiClassNMSKernel : public framework::OpKernel<T> {
 
       // Store the new indices.
       std::map<int, std::vector<int>> new_indices;
-      for (int j = 0; j < score_index_pairs.size(); ++j) {
+      for (size_t j = 0; j < score_index_pairs.size(); ++j) {
         int label = score_index_pairs[j].second.first;
         int idx = score_index_pairs[j].second.second;
         new_indices[label].push_back(idx);
@@ -238,7 +238,7 @@ class MultiClassNMSKernel : public framework::OpKernel<T> {
       int label = it.first;
       const T* sdata = scores_data + label * predict_dim;
       const std::vector<int>& indices = it.second;
-      for (int j = 0; j < indices.size(); ++j) {
+      for (size_t j = 0; j < indices.size(); ++j) {
         int idx = indices[j];
         const T* bdata = bboxes_data + idx * kBBoxSize;
         odata[count * kOutputDim] = label;           // label

